app/src/main/java/com/xinto/opencord/ui/viewmodel/CurrentUserViewModel.kt

```diff
@@ -7,26 +7,23 @@ import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.xinto.opencord.R
-import com.xinto.opencord.domain.manager.CacheManager
 import com.xinto.opencord.domain.mapper.toApi
-import com.xinto.opencord.domain.mapper.toDomain
 import com.xinto.opencord.domain.model.*
-import com.xinto.opencord.domain.repository.DiscordApiRepository
 import com.xinto.opencord.gateway.DiscordGateway
 import com.xinto.opencord.gateway.dto.UpdatePresence
-import com.xinto.opencord.gateway.event.ReadyEvent
-import com.xinto.opencord.gateway.event.SessionsReplaceEvent
-import com.xinto.opencord.gateway.event.UserSettingsUpdateEvent
-import com.xinto.opencord.gateway.event.UserUpdateEvent
-import com.xinto.opencord.gateway.onEvent
+import com.xinto.opencord.store.CurrentUserStore
+import com.xinto.opencord.store.SessionStore
+import com.xinto.opencord.store.UserSettingsStore
+import com.xinto.opencord.util.collectIn
 import com.github.materiiapps.partial.Partial
 import kotlinx.coroutines.launch
 import kotlinx.datetime.Clock
 
 class CurrentUserViewModel(
-    val repository: DiscordApiRepository,
-    val gateway: DiscordGateway,
-    val cache: CacheManager,
+    private val gateway: DiscordGateway,
```

Comment:
gateway is also unnecessary for the viewmodel since we do stores now

Reply:
Well here it is needed because we update presence here
I can't figure out where to move it because it depends on both user settings store and session store

Reply:
hm okay, we'll figure that out later then